PsfLauncher/main.cpp

```diff
@@ -268,43 +275,247 @@ void LaunchMonitorInBackground(std::filesystem::path packageRoot, const wchar_t
+        , static_cast<WORD>(cmdShow) // wShowWindow
+        }
+    };
 
-        PROCESS_INFORMATION processInfo;
+    if (runInAppContainer)
+    {
+        SIZE_T AttributeListSize;
+        InitializeProcThreadAttributeList(nullptr, 1, 0, &AttributeListSize);
+        startupInfoEx.lpAttributeList = (LPPROC_THREAD_ATTRIBUTE_LIST)HeapAlloc(
+            GetProcessHeap(),
+            0,
+            AttributeListSize
+        );
+
+        if (InitializeProcThreadAttributeList(startupInfoEx.lpAttributeList,
+            1,
+            0,
+            &AttributeListSize) == FALSE)
+        {
+            auto err{ ::GetLastError() };
+            std::wstring errorString = L"Could not initilize the proc thread attribute list.";
 
-        std::wstring cmdarg = cmd + L" " + arguments;
+            ErrorInformation error = { errorString, err };
+            return error;
+        }
 
-        if (!::CreateProcessW(
-            nullptr, //quotedapp.data(),
-            (wchar_t *)cmdarg.c_str(),
-            nullptr, nullptr, // Process/ThreadAttributes
-            true, // InheritHandles
-            0, // CreationFlags
-            nullptr, // Environment
+        DWORD attribute = PROCESS_CREATION_DESKTOP_APP_BREAKAWAY_DISABLE_PROCESS_TREE;
+        if (UpdateProcThreadAttribute(startupInfoEx.lpAttributeList,
+            0,
+            PROC_THREAD_ATTRIBUTE_DESKTOP_APP_POLICY,
+            &attribute,
+            sizeof(attribute),
             nullptr,
-            &startupInfo,
-            &processInfo))
+            nullptr) == FALSE)
         {
-            if (::GetLastError() == ERROR_ELEVATION_REQUIRED)
-                Log("error starting monitor using CreateProcessW. You must specify 'monitor/asadmin' in config.json\n");
-            else
-                Log("error starting monitor using CreateProcessW. Error=0x%x\n", ::GetLastError());
+            auto err{ ::GetLastError() };
+            std::wstring errorString = L"Could not update Proc thread attribute.";
+
+            ErrorInformation error{ errorString, err };
+            return error;
+        }
+    }
+
+    PROCESS_INFORMATION processInfo = { 0 };
+    if (::CreateProcessW(
+        execInfo.ApplicationName,
+        execInfo.CommandLine.data(),
+        nullptr, nullptr, // Process/ThreadAttributes
+        true, // InheritHandles
+        EXTENDED_STARTUPINFO_PRESENT, // CreationFlags
+        nullptr, // Environment
+        execInfo.CurrentDirectory,
+        (LPSTARTUPINFO)&startupInfoEx,
+        &processInfo))
+    {
+        // Propagate exit code to caller, in case they care
+        DWORD waitResult = ::WaitForSingleObject(processInfo.hProcess, INFINITE);
+
+        if (waitResult == WAIT_OBJECT_0)
+        {
+            ErrorInformation noError;
+            return noError;
         }
         else
         {
-            if (wait)
-                WaitForSingleObject(processInfo.hProcess, INFINITE);
+            auto err{ ::GetLastError() };

```

Comment:
>::GetLastError() [](start = 22, length = 16)

the comment indicates propagating the exit code, in which case you'd want this:
[https://docs.microsoft.com/en-us/windows/desktop/api/processthreadsapi/nf-processthreadsapi-getexitcodeprocess](https://docs.microsoft.com/en-us/windows/desktop/api/processthreadsapi/nf-processthreadsapi-getexitcodeprocess)

Reply:
I see.  Potentially, the wait could fail, but the process could have exited successfully and completed it's work, correct?